### PR TITLE
gh-119102: Fix REPL for dumb terminal

### PR DESCRIPTION
**Pull request by vstinner**
The site module gets the __main__ module to get _pyrepl.__main__.

* Issue: gh-119102

### CODE REVIEW

File: Lib/site.py [hunk 1]
```diff
@@ -523,7 +523,12 @@ def register_readline():
             pass
 
         def write_history():
-            from _pyrepl.__main__ import CAN_USE_PYREPL
+            try:
+                # _pyrepl.__main__ is executed as the __main__ module
+                from __main__ import CAN_USE_PYREPL
+            except ImportError:
+                CAN_USE_PYREPL = True
```
Comment (lysnikolaou) [inline Lib/site.py]: Shouldn't this probably be `False` if the import fails?

Comment (vstinner) [reply Lib/site.py]: I don't know. Let's use False on ImportError.

**Review by lysnikolaou — APPROVED** [review]
LGTM!

**Review by danielhollas — APPROVED** [review]
Works for me!

**Comment by vstinner** [comment]
Thanks for the review and testing!

**Pull request merged**